### ISSUE TITLE
ListResponse not passing check_response

### ISSUE THREAD

**Issue by ets** [yaal-coop/scim2-client#15]
Nice library - there's a real need for this and I'm excited to watch it develop.

Why is check_response attempting to validate the response payload using the Error schema? I'm using SCIMClient to test my service's endpoints and my validate ListResponse is being caught here resulting in an error. Is it simply the case that check_response shouldn't be used for endpoints that return a ListResponse?  Or am I missing the intent?

```python
        try:
            error = Error.model_validate(response_payload)
            if raise_scim_errors:
                raise SCIMResponseErrorObject(source=error)
            return error
        except ValidationError:
            pass
```

**Comment by azmeuk** [comment]
Thank you for your interest in the library!

The idea is that every SCIM endpoint can return an `Error` object, so the method first checks if the response payload is indeed an error object.
- If this is not an error object then the `ValidationError` is catched and the execution continues.
- If the payload is actually an error object, then depending on the `raise_scim_errors` parameter, it is either returned as the result of the method, or an exception is raised.

>  Is it simply the case that check_response shouldn't be used for endpoints that return a ListResponse?

It definitely can be used for endpoints returning a `ListResponse` payload. Can you share a sample of the problematic payload so I can better understand what is going on?

**Comment by azmeuk** [comment]
I think I can reproduce the behavior you are describing. It seems to be due to recent changes in scim2-models. I investigate and let you know.

**Comment by azmeuk** [comment]
This should be fixed with 0.1.7. Would you please check it solves your issues :pray:?

**Comment by ets** [comment]
Looks like it would.  I stopped using scim2-client for my testing due to a httpx dep conflict present in 0.1.6 but will try your latest soon.   Still using scim2-models for my FastAPI service though and have found that project extremely helpful.  Thanks for all your work!

**Comment by azmeuk** [comment]
FTR I released 0.1.8 which lowers the httpx minimum version to 0.24.0
We could probably lower the version again, but some unit tests break with 0.23.0, though they should not be too hard to fix.
